src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -0,0 +1,124 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent;
+import com.puppycrawl.tools.checkstyle.TreeWalkerFilter;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+
+/**
+ * This filter processes {@link TreeWalkerAuditEvent}
+ * objects based on the criteria of files, checks, message, module id, xpathQuery.
+ */
```

Comment:
unfortunately we have bad examples of javadoc over Filters....

Let not continue this practice in new filters we are creating. We need to do the same javadoc as we do for Checks. With majority common content with xdoc.
Eventually we will keep documentation only in javadoc and xdoc will be generated from javadoc.

Reply:
@romani So what should I do for this? Do I need to make it the same as the **description** in the documentation of SuppressionXpathSingleFilter?

Reply:
Yes

Reply:
done.